hazelcast/src/main/java/com/hazelcast/internal/diagnostics/OperationHeartbeatPlugin.java

```diff
@@ -0,0 +1,133 @@
+ * If there is a big deviation, then it could indicate networking problems. But it could also indicate other problems like JVM
+ * issues.
+ */
+public class OperationHeartbeatPlugin extends DiagnosticsPlugin {
+
+    /**
+     * The period in seconds this plugin runs.
+     *
+     * This plugin is very cheap to run and is enabled by default.
+     *
+     * It will also not print output every time it is executed, only when one or more members have a too high operation-heartbeat
+     * deviation, the plugin will render output.
+     *
+     * If set to 0, the plugin is disabled.
+     */
+    public static final HazelcastProperty PERIOD_SECONDS
+            = new HazelcastProperty(PREFIX + ".operation-heartbeat.seconds", 10, SECONDS);
+
+    /**
+     * The maximum allowed deviation. E.g. with a default 60 call timeout and operation-heartbeat interval being 15 seconds,
+     * the maximum deviation with a deviation-percentage of 33, is 5 seconds. So if a packet is arrived after 19 seconds; no
+     * problem; but if it arrives after 21 seconds, then the plugin will render.
+     */
+    public static final HazelcastProperty MAX_DEVIATION_PERCENTAGE
+            = new HazelcastProperty(PREFIX + ".operation-heartbeat.max-deviation-percentage", 33);
+
+    private static final float HUNDRED = 100f;
+
+    private final InvocationMonitor invocationMonitor;
+    private final long periodMillis;
+    private final long expectedIntervalMillis;
+    private final int maxDeviationPercentage;
+    private final ConcurrentMap<Address, AtomicLong> heartbeatPerMember;
+    private boolean mainSectionStarted;
+
+    public OperationHeartbeatPlugin(NodeEngineImpl nodeEngine) {
+        super(nodeEngine.getLogger(OperationHeartbeatPlugin.class));
+        this.invocationMonitor = ((OperationServiceImpl) nodeEngine.getOperationService()).getInvocationMonitor();
+        HazelcastProperties properties = nodeEngine.getProperties();
+        this.periodMillis = properties.getMillis(PERIOD_SECONDS);
+        this.maxDeviationPercentage = properties.getInteger(MAX_DEVIATION_PERCENTAGE);
+        this.expectedIntervalMillis = invocationMonitor.getHeartbeatBroadcastPeriodMillis();
+        this.heartbeatPerMember = invocationMonitor.getHeartbeatPerMember();
+    }
+
+    @Override
+    public long getPeriodMillis() {
+        return periodMillis;
+    }
+
+    @Override
+    public void onStart() {
+        logger.info("Plugin:active: period-millis:" + periodMillis + " max-deviation:" + maxDeviationPercentage + "%");
+    }
+
+    @Override
+    public void run(DiagnosticsLogWriter writer) {
+        long nowMillis = System.currentTimeMillis();
+        for (Map.Entry<Address, AtomicLong> entry : heartbeatPerMember.entrySet()) {
+            Address member = entry.getKey();
+            long lastHeartbeatMillis = entry.getValue().longValue();
```

Comment:
this can return `0` for new members -> the plugin will render absurdly high deviation. 
this is from the InvocationMonitor:
```java
        private void updateMemberHeartbeat(long timeMillis) {
            AtomicLong lastMemberHeartbeat = lastHeartbeatPerMember.get(sender);
            if (lastMemberHeartbeat == null) {
                lastMemberHeartbeat = new AtomicLong();
                lastHeartbeatPerMember.put(sender, lastMemberHeartbeat);
            }

            lastMemberHeartbeat.set(timeMillis);
        }
```